core/src/main/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunction.java

```diff
@@ -500,26 +500,30 @@ private static DefaultFunctionResolver truncate() {
         FunctionDSL.impl(
             FunctionDSL.nullMissingHandling(
                 (x, y) -> new ExprLongValue(
-                    new BigDecimal(x.integerValue()).setScale(y.integerValue(),
-                        RoundingMode.DOWN).longValue())),
+                        BigDecimal.valueOf(x.integerValue()).setScale(y.integerValue(),
+                                x.integerValue() > 0 ? RoundingMode.FLOOR : RoundingMode.CEILING)
```

Comment:
After checking the Javadoc, I wonder should we use `RoundingMode.DOWN` for both cases?

Because it says:

```
   /**
    * Rounding mode to round towards negative infinity. If the result is positive, behave as for
    * RoundingMode.DOWN ...
    */
    FLOOR(BigDecimal.ROUND_FLOOR),

   /**
    * Rounding mode to round towards positive infinity. ... if negative, behaves as for
    * RoundingMode.DOWN
    */
    CEILING(BigDecimal.ROUND_CEILING),
```

Reply:
Addressed in 74c575a1d8ee91f902f0002cc878a75601ad7fdb

Reply:
> Addressed in [74c575a](https://github.com/opensearch-project/sql/commit/74c575a1d8ee91f902f0002cc878a75601ad7fdb)

Thanks. But I think we can change line 511, 518, 525 too?

Reply:
Changed in 7a5b5a65bb6bd32fa29136219d632b7a234b6976